src/player.cpp

```diff
@@ -10186,19 +10185,14 @@ int player::get_env_resist( body_part bp ) const
         }
     }
 
-    if( bp == bp_mouth && has_bionic( bio_purifier ) && ret < 5 ) {
-        ret += 2;
-        if( ret > 5 ) {
-            ret = 5;
+    for( const bionic &bio : *my_bionics ) {
+        for( const auto &element : bio.info().env_protec ) {
```

Comment:
You have a `std::map` here, so you shouldn't need to loop over it.  Although it would be a little awkward to do the lookup if you keep the eyes special case code.